src/py/pyodide/code.py

```diff
@@ -25,11 +26,80 @@ def run_js(code: str, /) -> Any:
     return eval_(code)
 
 
+def _relaxed_call_sig(func):
+    from inspect import Parameter, signature
+
+    try:
+        sig = signature(func)
+    except (TypeError, ValueError):
+        return None
+    new_params = list(sig.parameters.values())
+    idx: int | None = -1
+    for idx, param in enumerate(new_params):
+        if param.kind in (Parameter.KEYWORD_ONLY, Parameter.VAR_KEYWORD):
+            break
+        if param.kind == Parameter.VAR_POSITIONAL:
+            idx = None
+            break
+    else:
+        idx += 1
+    if idx is not None:
+        new_params.insert(idx, Parameter("__var_positional", Parameter.VAR_POSITIONAL))
+
+    for param in new_params:
+        if param.kind == Parameter.KEYWORD_ONLY:
+            break
+    else:
+        new_params.append(Parameter("__var_keyword", Parameter.VAR_KEYWORD))
+    new_sig = sig.replace(parameters=new_params)
+    return new_sig
+
+
+@lru_cache
+def _relaxed_call_sig_cached(func):
```

Comment:
Question: Why not put cache decorator directly to `_relaxed_call_sig`?

Reply:
Oh I meant to call the uncached `_relaxed_call_sig` from `relaxed_wrap`. I imagine `relaxed_call` will be called repeatedly with the same function argument but `relaxed_wrap` only once.